opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2739,23 +2736,24 @@ void CpGrid::populateLeafGridFaces(Dune::cpgrid::EntityVariableBase<cpgrid::Geom
         // Note: Since we are associating each LGR with its parent cell index, and this index can take
         //       the value 0, with the value -1 we refer to current_view_data_ (grid where the elements have been marked).
         const auto& elemLgrFaceEntity =  Dune::cpgrid::EntityRep<1>(elemLgrFace, true);
+        const auto& grid_or_elemLgr_data = (elemLgr == -1) ? *current_view_data_ : *markedElem_to_itsLgr[elemLgr];
 
         // Get the face geometry.
-        adapted_faces[face] = (*( ((elemLgr == -1) ? *current_view_data_ : *markedElem_to_itsLgr[elemLgr]).geometry_.geomVector(std::integral_constant<int,1>()) ) )[elemLgrFaceEntity];
+        adapted_faces[face] = (*(grid_or_elemLgr_data.geometry_.geomVector(std::integral_constant<int,1>())))[elemLgrFaceEntity];
         // Get the face tag.
-        mutable_face_tags[face] = ((elemLgr == -1) ? *current_view_data_ : *markedElem_to_itsLgr[elemLgr]).face_tag_[elemLgrFaceEntity];
+        mutable_face_tags[face] = grid_or_elemLgr_data.face_tag_[elemLgrFaceEntity];
         // Get the face normal.
-        mutable_face_normals[face] = ((elemLgr == -1) ? *current_view_data_ : *markedElem_to_itsLgr[elemLgr]).face_normals_[elemLgrFaceEntity];
+        mutable_face_normals[face] = grid_or_elemLgr_data.face_normals_[elemLgrFaceEntity];
         // Get face_to_point_ before adapting - we need to replace the level corners by the adapted ones.
-        Opm::SparseTable<int>::mutable_row_type preAdapt_face_to_point = ((elemLgr == -1) ? *current_view_data_ : *markedElem_to_itsLgr[elemLgr]).face_to_point_[elemLgrFace];
+        const auto& preAdapt_face_to_point = grid_or_elemLgr_data.face_to_point_[elemLgrFace];
         // Add the amount of points to the count num_points.
         num_points += preAdapt_face_to_point.size();
 
         for (int corn = 0; corn < static_cast<int>(preAdapt_face_to_point.size()); ++corn) {
             int adaptedCorn = 0; // It'll get rewritten.
             const auto& elemLgrCorn = preAdapt_face_to_point[corn];
             // Corner is stored in adapted_corners
-            if ( elemLgrAndElemLgrCorner_to_adaptedCorner.count({elemLgr, elemLgrCorn}) == 1)  {
+            if ( elemLgrAndElemLgrCorner_to_adaptedCorner.find({elemLgr, elemLgrCorn}) !=  elemLgrAndElemLgrCorner_to_adaptedCorner.end()) {
```

Comment:
Please store iterator returned by find and use that to get the value. see other comment.